.travis.yml

```diff
@@ -49,7 +49,7 @@ before_script:
   - adb shell input keyevent 82 &
 
 script:
-  - ./gradlew build connectedCheck
+  - ./gradlew connectedTravisDebugAndroidTest
```

Comment:
Product flavors take advantage of Gradle's slippery nature.  

So connectedCheck runs the default tests on an emulator.  To use the Travis build flavor and run the Travis tests I use connectedTravisDebugAndroidTest.  It is a bunch of properties dynamically chained together in the task name.